### ISSUE TITLE
[require-template]: false negative to TypeScript overload function

### ISSUE THREAD

**Issue by otomad** [gajus/eslint-plugin-jsdoc#1462]
## Expected behavior

It should report error to type parameters of the overloaded functions when jsdoc doesn't declare `@template`.

## Actual behavior

Nothing reported.

## ESLint Config

```js
export default [
	tseslint.configs.base,
	{
		files: ["**/*.{js,ts}"],
		plugins: {
			jsdoc,
		},
		rules: {
			"jsdoc/require-template": ["error", {
				"requireSeparateTemplates": true,
			}],
		},
	},
];
```

## ESLint sample

This will not report any error.

```ts
/**
 * @param bar
 * @param baz
 * @returns
 */
function foo<T>(bar: T, baz: number): T;
function foo<T>(bar: T, baz: boolean): T;
function foo<T>(bar: T, baz: number | boolean): T {
	return bar;
}
```

This will report the error properly.

```ts
function foo<T>(bar: T, baz: number): T;
function foo<T>(bar: T, baz: boolean): T;
/**
 * @param bar
 * @param baz
 * @returns
 */
function foo<T>(bar: T, baz: number | boolean): T {
	return bar;
}
```

## Environment

- Node version: v23.6.1
- ESLint version v9.23.0
- `eslint-plugin-jsdoc` version: Test both v54.1.1 and #1452
